src/main/java/org/jboss/xavier/integrations/route/model/PageResponse.java

```diff
@@ -0,0 +1,102 @@
+package org.jboss.xavier.integrations.route.model;
+
+import java.util.List;
+
+public class PageResponse<T> {
```

Comment:
@jonathanvila I don't think there is something we can do since ApiService does not have a way to know the Class the Page is using. Is there anything special you have in mind?

Reply:
Well, I see it's a good idea to use Generics.
So, ApiService could have a method like this 
```
     public <T> PageResponse<T> mapToCustomPage(Exchange exchange, Class<T> expectedClass) {
``` and then we would also have the real object, instead of Object.

Reply:
@jonathanvila this would imply to change the way we call `mapToCustomPage`. For instance we should need to add an extra parameter in camel-context.xml 

```
<bean ref="apiService" method="mapToCustomPage(${exchange}, CLASS_TYPE)"/>
```

I suggest to keep the current version of ApiService so we continue calling the service using:
```
<bean ref="apiService" method="mapToCustomPage(${exchange})"/>
```

Reply:
why ? which is the benefit or drawback you see here ?
I would say that adding or not adding an extra param doesnt sound like a hard reason....